x/paramfilter/gov_handler.go

```diff
@@ -0,0 +1,77 @@
+package paramfilter
+
+import (
+	"fmt"
+
+	sdkerrors "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	legacysdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
+	paramskeeper "github.com/cosmos/cosmos-sdk/x/params/keeper"
+	"github.com/cosmos/cosmos-sdk/x/params/types/proposal"
+)
+
+// ParamBlockList keeps track of parameters that cannot be changed by governance
+// proposals
+type ParamBlockList struct {
+	forbiddenParams map[string]bool
+}
+
+// NewParamBlockList creates a new ParamBlockList that can be used to block gov
+// proposals that attempt to change locked parameters.
```

Comment:
```suggestion
// proposals that attempt to change hard-coded parameters.
```

Reply:
updated, but the reasoning for not using hard-coded before was because I still don't consider these to be hard coded despite them not being changable. imo, a hard-coded value is something that is a literal constant in the code, not a protected parameter